Update Workers Analytics Engine docs (#13712)
cloudflare/cloudflare-docs: content/analytics/analytics-engine/get-started.md

dataset = "<DATASET_NAME>"
```

## 2. Write data points from your Worker

You can write data points to your Worker by calling the `writeDataPoint()` method that is exposed on the binding that you just created.

```js
async fetch(request, env) {
  env.WEATHER.writeDataPoint({
    'blobs': ["Seattle", "USA", "pro_sensor_9000"], // City, State
    'doubles': [25, 0.5],
    'indexes': ["a3cd45"]
  });
  return new Response("OK!");
}
```

{{<Aside type="note">}}
You do not need to await `writeDataPoint()` — it will return immediately, and the Workers runtime handles writing your data in the background.
{{</Aside>}}

A data point is a structured event that consists of:

> - **Blobs** (numbers) — the numeric values that you want to record in your data point.

Reviewer: ```suggestion
- **Blobs** (numbers) — The numeric values that you want to record in your data point.
```

> - **Doubles** (strings) — the dimensions used for grouping and filtering. Sometimes called labels in other metrics systems.

Reviewer: ```suggestion
- **Doubles** (strings) — The dimensions used for grouping and filtering. Sometimes called labels in other metrics systems.
```

> - **Indexes** — (strings) — used as a [sampling](/analytics/analytics-engine/sql-api/#sampling) key.

Reviewer: ```suggestion
- **Indexes** — (strings) — Used as a [sampling](/analytics/analytics-engine/sql-api/#sampling) key.
```

In the example above, suppose you are collecting air quality samples. Each data point written represents a reading from your weather sensor. The blobs define city, state, and sensor model — the dimensions you want to be able to filter queries on later. The doubles define the numeric temperature and air pressure readings. And the index is the ID of your customer. You may want to include [context about the incoming request](/workers/runtime-apis/request/), such as geolocation, to add additional data to your datapoint.

Currently, the `writeDataPoint()` API accepts ordered arrays of values. This means that you must provide fields in a consistent order.

## 3. Query data using the SQL API

You can query the data you have written in two ways:

> - [**SQL API**](/analytics/analytics-engine/sql-api) — best for writing your own queries and integrating with external tools like Grafana.

Reviewer: ```suggestion
- [**SQL API**](/analytics/analytics-engine/sql-api) — Best for writing your own queries and integrating with external tools like Grafana.
```

> - [**GraphQL API**](/analytics/graphql-api/) — this is the same API that powers the Cloudflare dashboard.

Reviewer: ```suggestion
- [**GraphQL API**](/analytics/graphql-api/) — This is the same API that powers the Cloudflare dashboard.
```

> For the purpose of this example, we'll use the SQL API.

Reviewer: ```suggestion
For the purpose of this example, we will use the SQL API.
```

### Create an API token

Create an [API Token](https://dash.cloudflare.com/profile/api-tokens) that has the `Account Analytics Read` permission.

### Write your first query

The following query returns the top 10 cities that had the highest average humidity readings when the temperature was above zero:

```sql
SELECT
  blob1 AS city,
  SUM(_sample_interval * double2) / SUM(_sample_interval) AS avg_humidity
FROM WEATHER
WHERE double1 > 0
GROUP BY city
ORDER BY avg_humidity DESC
LIMIT 10
```

{{<Aside type="note">}}
We are using a custom averaging function to take [sampling](/analytics/analytics-engine/sql-api/#sampling) into account.
{{</Aside>}}

You can run this query by making an HTTP request to the SQL API:

```curl
curl -X POST "https://api.cloudflare.com/client/v4/accounts/YOUR_ACCOUNT_ID/analytics_engine/sql" -H "Authorization: Bearer YOUR_API_TOKEN" -d "SELECT blob1 AS city, SUM(_sample_interval * double2) / SUM(_sample_interval) AS avg_humidity FROM WEATHER WHERE double1 > 0 GROUP BY city ORDER BY avg_humidity DESC LIMIT 10"
```

> Refer to the the [Workers Analytics Engine SQL Reference](/analytics/analytics-engine/sql-reference/) for a full list of supported SQL functionality.

Reviewer: ```suggestion
Refer to the [Workers Analytics Engine SQL Reference](/analytics/analytics-engine/sql-reference/) for a full list of supported SQL functionality.
```

Refer to [Querying Workers Analytics Engine from Grafana](/analytics/analytics-engine/grafana/) for more details on how to create efficient Grafana queries against Workers Analytics Engine.

## Further reading

> {{<directory-listing folderDirectory="/analytics/analytics-engine/" showDescriptions="true">}}

Reviewer 2: ```suggestion
{{<directory-listing folderDirectory="/analytics/analytics-engine/">}}
```

Reviewer 2: This is the issue. Weird. I'll file a bug ticket to look into this.
